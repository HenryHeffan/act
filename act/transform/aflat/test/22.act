defproc and22 (bool a[2][2]; bool c[2]){
	prs{
		~a[0][0] | ~a[0][1] -> c[0]+
		 a[0][0] &  a[0][1] -> c[0]-
		~a[1][0] | ~a[1][1] -> c[1]+
		 a[1][0] &  a[1][1] -> c[1]-	
	}
}

defproc foo(bool a22[4][4], c2[4]) {
  and22 and22[2];
  (i:2: and22[i](a22[2*i..2*i+1][2*i..2*i+1], c2[2*i..2*i+1]);)
}

foo x;
