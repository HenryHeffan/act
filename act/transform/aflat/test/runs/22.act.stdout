~"x.and22[0].a[0][0]"|~"x.and22[0].a[0][1]"->"x.and22[0].c[0]"+
"x.and22[0].a[0][0]"&"x.and22[0].a[0][1]"->"x.and22[0].c[0]"-
~"x.and22[0].a[1][0]"|~"x.and22[0].a[1][1]"->"x.and22[0].c[1]"+
"x.and22[0].a[1][0]"&"x.and22[0].a[1][1]"->"x.and22[0].c[1]"-
~"x.and22[1].a[0][0]"|~"x.and22[1].a[0][1]"->"x.and22[1].c[0]"+
"x.and22[1].a[0][0]"&"x.and22[1].a[0][1]"->"x.and22[1].c[0]"-
~"x.and22[1].a[1][0]"|~"x.and22[1].a[1][1]"->"x.and22[1].c[1]"+
"x.and22[1].a[1][0]"&"x.and22[1].a[1][1]"->"x.and22[1].c[1]"-
= "x.c2[0]" "x.and22[0].c[0]"
= "x.c2[1]" "x.and22[0].c[1]"
= "x.c2[2]" "x.and22[1].c[0]"
= "x.c2[3]" "x.and22[1].c[1]"
= "x.a22[0][0]" "x.and22[0].a[0][0]"
= "x.a22[0][1]" "x.and22[0].a[0][1]"
= "x.a22[1][0]" "x.and22[0].a[1][0]"
= "x.a22[1][1]" "x.and22[0].a[1][1]"
= "x.a22[2][2]" "x.and22[1].a[0][0]"
= "x.a22[2][3]" "x.and22[1].a[0][1]"
= "x.a22[3][2]" "x.and22[1].a[1][0]"
= "x.a22[3][3]" "x.and22[1].a[1][1]"
